### PR TITLE
added new param target_site to datacheck to filter metakeys

### PR DESCRIPTION
**Pull request by vinay-ebi**
New param target_site added to datacheck script & pipeline to filter the meta keys from production db based on this new filed

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm [hunk 1]
```diff
@@ -36,22 +36,27 @@ use constant {
...
   /;
+
   my $helper = $self->dba->dbc->sql_helper;
   my %meta_keys = %{ $helper->execute_into_hash(-SQL => $sql) };
 
+  #check target site is main / new and select mandatory metakeys 
+  my $filter_metakeys = '';
+  if (defined $self->target_site){
+    $filter_metakeys = " AND target_site like '\%new\%' ";
+  }
```
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm:54]: I believe the filter should use the para value:
```suggestion
  if (defined $self->target_site){
    $filter_metakeys = " AND target_site like '\%".$self->target_site."\%' ";
  }
```

File: scripts/run_datachecks.pl [hunk 2]
```diff
@@ -154,7 +159,7 @@ =head1 OPTIONS
     $host, $port, $user, $pass, $dbname, $dbtype,
     $registry_file, @server_uri, @old_server_uri, $data_file_path, $config_file,
     @names, @patterns, @groups, @datacheck_types,
-    $datacheck_dir, $index_file, $history_file, $output_file,
+    $datacheck_dir, $index_file, $history_file, $output_file,, $target_site
```
Comment (marcoooo) [inline scripts/run_datachecks.pl:162]: Isn't there a double `,,` here?

**Comment by marcoooo** [comment]
Is this the only impacted Check?  Sure 100%?

**Review by marcoooo — CHANGES_REQUESTED** [review]

**Comment by vinay-ebi** [comment]
> Is this the only impacted Check? Sure 100%?

the current test specifically accesses the 'ensembl_production' database to compare meta keys. It is important to note that other DC listed below will not perform a meta keys comparison in the production database meta_key table:

CompareMetaKeys
MetaCoord
MetaKeyAssembly
MetaKeyCardinality
MetaKeyConditional
MetaKeyConsistent
MetaKeyFormat
MetaKeyLevel

 Ran all core and rapid_release group dc on `zea_mays_core_57_110_8` on st6 and results are attached in Jira ticket

File: lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm [hunk 3]
```diff
@@ -36,22 +36,27 @@ use constant {
...
     GROUP BY meta_key
   /;
+
   my $helper = $self->dba->dbc->sql_helper;
   my %meta_keys = %{ $helper->execute_into_hash(-SQL => $sql) };
 
+  #check target site is main / new and select mandatory metakeys 
+  my $filter_metakeys = '';
+  if (defined $self->target_site){
+    $filter_metakeys = " AND target_site like '\%".$self->target_site."\%' ";
```
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm:53]: Another question, sorry I missed it last time, why escaping the `%` in the string?

Comment (dpopleton) [reply lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm:53]: I don't think they will hurt, but I don't see them doing anything in this context.

Comment (vinay-ebi) [reply lib/Bio/EnsEMBL/DataCheck/Checks/ControlledMetaKeys.pm:53]: with in double quote it interpolates as hash, escaped this character.

**Comment by vinay-ebi** [comment]
https://github.com/Ensembl/ensembl-datacheck/pull/568

**Pull request closed without merging**